config/src/main/java/com/typesafe/config/impl/ResolveSource.java

```diff
@@ -242,9 +242,7 @@ ResolveSource replaceWithinCurrentParent(AbstractConfigValue old, AbstractConfig
             if (old == root && replacement instanceof Container) {
                 return new ResolveSource(rootMustBeObj((Container) replacement));
             } else {
-                throw new ConfigException.BugOrBroken("replace in parent not possible " + old + " with " + replacement
-                        + " in " + this);
-                // return this;
+                return this;
```

Comment:
so I'm trying to reload this in my brain, I wonder if this comment would be accurate
```suggestion
                // this happens when we're trying to do a replacement in a source that
                // isn't our ancestor, so there's no point doing a replacement at all.
                return this;
```